mozjs/build/moz.configure/rust.configure

```diff
@@ -92,6 +92,8 @@ def rust_target(rust_compiler, rustc, target, cross_compiling):
             # Android
             ('x86', 'Android'): 'i686-linux-android',
             ('arm', 'Android'): 'arm-linux-androideabi',
+            ('aarch64', 'Android'): 'aarch64-linux-android',
+            ('armv7', 'Android'): 'arm-linux-androideabi',
```

Comment:
The other changes in this PR will need to be [submitted upstream](https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=JavaScript%20Engine) if they're not [already present](https://hg.mozilla.org/mozilla-central/file/38f862749aed/js).